pkg/agents/team2/config.go

```diff
@@ -77,58 +73,111 @@ func InitTable(numStates int, numActions int) [][]float64 {
-		a.isPlatArrived = false
-	}
-	if a.isPlatArrived {
+	//Perform the following only once per day when platform arrives
+	//a.exportQTable()
+	//a.exportPolicies()
+
+	if a.PlatformOnFloor() && a.isNewDay() {
 		oldState := a.CheckState()
 		oldHP := a.HP()
 		a.Log("Agent team2 before action:", infra.Fields{"floor": a.Floor(), "hp": oldHP, "food": a.CurrPlatFood(), "state": oldState})
 		action := a.SelectAction()
-		_, err := a.TakeFood(a.actionSpace.actionSet[action](oldHP)) //perform selected action
+
+		foodTaken, err := a.TakeFood(food.FoodType(a.actionSpace[action])) //perform selected action
+		a.lastFoodTaken = foodTaken
 		if err != nil {
-			switch err.(type) {
-			case *infra.FloorError:
-			case *infra.NegFoodError:
-			case *infra.AlreadyEatenError:
-			default:
-			}
+			//if there's error, cease updating tables
+			return
 		}
 		a.Log("Agent team2:", infra.Fields{"selected and performed action": action})
 		a.Log("Agent team2 after action:", infra.Fields{"floor": a.Floor(), "hp": a.HP(), "food": a.CurrPlatFood(), "state": a.CheckState()})
+		if a.DaysAtCritical() > 0 && a.PreviousdayAtCritical != a.DaysAtCritical() {
+			a.PreviousdayAtCritical = a.DaysAtCritical()
+			a.daysAtCriticalCounter += 1
+			if a.DaysAtCritical() >= (a.HealthInfo().MaxDayCritical - 1) {
+				a.Log("Agent team2 at critical state", infra.Fields{"daysAtCriticalCounter": a.daysAtCriticalCounter, "floor": a.Floor(), "hp": a.HP(), "food": a.CurrPlatFood(), "state": a.CheckState()})
+			}
+		}
+		a.ReplyAllAskMsg()
 		hpInc := a.HP() - oldHP
-		a.updateRTable(hpInc, oldState, action)
+		a.updateRTable(oldHP, hpInc, oldState, action)
 		a.updateQTable(oldState, action)
-		a.updatePolicies(oldState)
+		a.updatePolicies(oldState, action)
+		a.lastAge = a.Age()
+	}
+
+}
+
+func (a *CustomAgent2) isNewDay() bool {
+	if a.Age() < a.lastAge {
+		a.lastAge = -1
+		return true
+	}
+	if a.Age() == a.lastAge {
+		return false
 	}
+	return true
+}
+
+func (a *CustomAgent2) HandleAskHP(msg messages.AskHPMessage) {
+	reply := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), a.HP())
+	a.SendMessage(reply)
+	a.Log("Team2 replying askHP message from ", infra.Fields{"senderFloor": msg.SenderFloor(), "myFloor": a.Floor()})
+}
+
+func (a *CustomAgent2) HandleAskFoodTaken(msg messages.AskFoodTakenMessage) {
+	reply := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), int(a.lastFoodTaken))
+	a.SendMessage(reply)
+	a.Log("Team2 replying askFoodTaken message", infra.Fields{"senderFloor": msg.SenderFloor(), "myFloor": a.Floor()})
+}
+
+func (a *CustomAgent2) HandleAskIntendedFoodTaken(msg messages.AskIntendedFoodIntakeMessage) {
+	reply := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), (a.lastAction+1)*5)
+	a.SendMessage(reply)
+	a.Log("Team2 replying askIntendedFoodTaken message from ", infra.Fields{"senderFloor": msg.SenderFloor(), "myFloor": a.Floor()})
+}
+
+func (a *CustomAgent2) HandleStateHP(msg messages.StateHPMessage) {
+	statement := msg.Statement()
+	a.neiboughHP = statement
```

Comment:
This neighbour HP variable will be updated regardless of who sends the message even if it s from multiple floors above, is that intended?

Reply:
Thank you for pointing it out! Fixed it.